test/integration/binary/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+
+if (SWIG_PYTHON OR SWIG_JAVA)
+  get_property(SWIG_BUILD_DIR GLOBAL PROPERTY SWIG_BUILD_DIR)
+endif ()
+
+if (SWIG_PYTHON)
+  if (SUPPORT_PYTHON2)
+    find_package(PythonInterp 2.7 REQUIRED)
+  else ()
+    find_package(PythonInterp 3.5 REQUIRED)
+  endif ()
+
+  foreach (item "block" "commands" "primitive" "queries")
+    compile_proto_to_python("${item}.proto")
+    list(APPEND PROTO_SWIG_DEPS "${SWIG_BUILD_DIR}/${item}_pb2.py")
+  endforeach (item)
+
+  add_executable(binary_test
+      launchers.cpp
+      binaries_test.cpp)
+  target_link_libraries(binary_test
+      gtest
+      gmock
+      boost
+      schema
+      shared_model_proto_backend
+      shared_model_interfaces
+      integration_framework
+      bindings
+      )
+  target_include_directories(binary_test PUBLIC ${PROJECT_SOURCE_DIR}/test)
+
+  add_dependencies(binary_test irohapy)
+  add_test(
```

Comment:
Use `addtest` instead. https://github.com/hyperledger/iroha/blob/e8cdcc1f0b87058814e8d44c1b5f7b1839af3e92/cmake/functions.cmake#L23

Pls, note that function assumes "type" of test which is required for CI.

Reply:
It is not possible, because a custom test is defined here, while `addtest` creates a generic test which calls the test binary without any parameters. The test call is replaces here with python interpreter call.

Reply:
Can we create a generic function for those purposes? We have to manage types of tests in CI.

Reply:
Maybe an easier solution would be to override `add_test`, do custom logic and pass the arguments to real `add_test`. This will add types to all tests added with `add_test`.

Reply:
Discussed offline. Summary: there is no problem because `python_binary_test` target becomes visible only when SWIG for Python is enabled (`-DSWIG_PYTHON=ON`). Thus, CI would not be confused.